include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4520,6 +4520,15 @@ struct OrtApi {
    * \since Version 1.17.
    */
   ORT_API2_STATUS(ReadOpAttr, _In_ const OrtOpAttr* op_attr, _In_ OrtOpAttrType type, _Inout_ void* data, _In_ size_t len, _Out_ size_t* out);
+
+  /** \brief Used for custom operators, set an output of a kernel
+   *
+   * \see ::OrtCustomOp
+   *
+   * \since Version 1.17.
+   */
+  ORT_API2_STATUS(KernelContext_SetOutput, _Inout_ OrtKernelContext* context, _In_ size_t index,
```

Comment:
>KernelContext_SetOutput [](http://example.com/codeflow?start=18&length=23)

why we want to expose it to CAPI? we have TRT based custom op?

Reply:
> > KernelContext_SetOutput
> 
> why we want to expose it to CAPI? we have TRT based custom op?

compile api based EP's need to implement compute_func which only has access to public OrtKernelContext api , not the internal OpKernelContext api.  
we need to use SetOutputMLValue() so that's why it's plumbed thru to the public api in this PR